[BB] 리뷰 PR관련 리팩토링 - Modal, Card

BB-choi opened the pull request:
- modal 핸들러 함수 위치 변경
    - App에서 state로 관리하도록 변경했습니다.
        - columns를 App에서 생성하므로 App에서 state로 관리
        - 선택된id를 state로 관리하므로 re-rendering으로 인해 클래스 토글이 아닌 state로 모달 표시 여부 결정하였습니다.
- 기타사항 : dateUtils사용하여 todos 데이터를 최신순으로 정렬

와..! 이제 마지막날이네요 파이팅

BB-choi committed:
refactor: Card.js 불필요한 함수 삭제

- onMouseOver, onMouseOut 등 불필요한 함수 삭제

BB-choi committed:
feat: 카드 삭제 / 모달 팝업 닫을 때 선택된 id 리셋

BB-choi committed:
fix: x 버튼 hover시 cursor 범위 오류 해결

healtheloper commented on front/src/App.js:
```diff
@@ -36,7 +45,29 @@ const App = () => {
+    setSelectedTodoId,
+    handleModalVisibility,
+  };
+
```
오.. handlers..
setSelectedTodoId 도 handle 함수로 별도로 만들어서 써도 좋겠다는 생각이네요 `handle~` 라는 이름이 통일되게 👍
selectedTodoId 는 값이니까 isModalVisible 처럼 따로 넘겨주고요,
아이디어 아주 좋네용

healtheloper commented on front/src/App.js:
```diff
@@ -36,7 +45,29 @@ const App = () => {
+
+  const columnsHandlers = {
+    setSelectedTodoId,
+    handleModalVisibility,
```
이 아이도 마찬가지로?ㅎㅎ

healtheloper commented on front/src/components/Content/Cards/Cards.js:
```diff
@@ -1,20 +1,22 @@
   };
 
-  const todoElements = todos.sort(setTodosSortByLatest).map(getCardsTemplate);
+  const todoElements = getDataSortbyLatest(todos).map(createCardElement);
```
getSortedDataLatest ?

healtheloper commented on front/src/components/Content/Columns/Columns.js:
```diff
@@ -20,7 +20,7 @@ const Columns = ({ columns, todos, handleRenderFlag }) => {
       todos: getTodosByColumnId(column._id),
-      handleRenderFlag,
+      handlers,
     });
```
👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍

healtheloper commented on front/src/components/Modal/Modal.js:
```diff
@@ -61,11 +62,14 @@ const Modal = ({ todoId, handleRenderFlag }) => {
-    className: styles.modalWrap,
+    className: isModalVisible
+      ? [styles.modalWrap, styles.visible]
+      : [styles.modalWrap],
```
```js
[1, 2, ...(false ? [3] : [])]
```
이런 것도 가능하긴 한데, 비비 생각은 어떤 게 더 좋아보이나요?ㅎㅎ

healtheloper reviewed (changes requested):
아주 좋네요 👍👍👍👍👍

조금 변경해볼만한 것 같이 리뷰 드렸어요

BB-choi committed:
refactor: Cards.js에서 todos 정렬 함수 외부로 분리

- dateUtils 사용하여 분리

BB-choi committed:
refactor: modal 핸들러 함수 위치 변경

- App에서 state로 관리하도록 변경
    - columns를 App에서 생성하므로 App에서 state로 관리
    - 선택된id를 state로 관리하므로 re-rendering으로 인해 클래스 토글이
      아닌 state로 모달 표시 여부 결정

BB-choi committed:
fix: PR confilct 해결

BB-choi committed:
refactor: SideContent 최신순 정렬

BB-choi committed:
refactor: PR리뷰 반영
